api/docs/scatter_gather_emulation.dox

```diff
@@ -120,16 +239,23 @@ num_accesses = vector_size / element_size
 for i = 0, 1, ..., (num_accesses-1), do
   extract mask for the ith access from mask reg or mask vector
   if mask is set, then
-    extract ith element of index vector
-    compute address = base + ith index element
+    if index is vector, then
+        extract ith element of index vector
+        compute address = base + ith index element
+    else // base is vector
+        extract ith element of base vector
+        compute address = ith base element + index
+    done
     if instr_is_gather, then
       load data from address into a scalar reg
       insert scalar data into destination vector
     else // instr_is_scatter
       extract scalar data from source vector to scalar reg
       store data from scalar reg to address
     done
-    clear ith mask in mask reg or mask vector
+    if x86, then
+        clear ith mask in mask reg or mask vector
```

Comment:
Doesn't aarch64 do the same to the predicate reg? If some operation faults, what would the predicate reg show?

Reply:
No. The governing predicate register is never modified by any of the scatter/gather instructions, whether there is a fault or not.
For first-fault and non-fault loads there is a special first-fault predicate register (FFR) which is modified (I have added details about that to this doc)

Reply:
I believe on x86 a scatter/gather can be interrupted in the middle and resumed and that's what the mask reg update is for: so the resumption point is right there in the arch state.  Kind of like x86's `rep movs` and other string loops where the remaining trip count is in a register making it easy to restart from the middle.

On a64 can a scatter/gather not be interrupted in the middle and then resumed?  Is that why the faulting behavior is the way it is, zeroing out the future operations on a fault?